http_general/test_block_action.py

```diff
@@ -85,13 +106,30 @@ def test_block_action_attack_reply(self):
         self.save_must_fin_socks([client])
         self.save_must_not_reset_socks([client])
 
-        client.send_request(
+        client.make_request(
             request=f"GET / HTTP/1.1\r\nHost: bad.com\r\n\r\n",
-            expected_status_code="403",
         )
+        self.check_last_error_response(client, expected_status_code="403")
+
+        self.assertTrue(client.wait_for_connection_close())
+        self.check_fin_no_rst_in_sniffer(sniffer)
+
+    def test_block_action_error_reply_with_conn_close(self):
+        client = self.get_client("deproxy")
+
+        sniffer = self.setup_sniffer()
+        self.start_all_services()
+        self.save_must_fin_socks([client])
+        self.save_must_not_reset_socks([client])
+
+        client.send_request(
+            request=f"GET / HTTP/1.1\r\nHost: good.com\r\nContent-Type: invalid\r\n\r\n",
+            expected_status_code="400",
```

Comment:
Why do we use a string value for HTTP response status codes? It looks like almost everywhere including the built-in python libraries uses integer for HTTP response status codes (example `http.HTTPStatus`). And, if it is necessary, perhaps it would be better and clearer for this purpose to use a custom Enum class with the values of HTTP response status codes.

Reply:
I agree with you. But we have status as string [here](https://github.com/tempesta-tech/tempesta-test/blob/13762e32fd66206dde7d7d3cb9c2695da4587c18/helpers/deproxy.py#L620). I think you can fix this as part of #56